### PR TITLE
Add 'native' to -C target-cpu=help

### LINKED ISSUE #44393: rustc doesn't list `native` as a cpu option
## short summary of the bug

```
$ rustc -Ctarget-cpu=help
```

Doesn't list the `native` as an option.

## I expected to see this happen: 

I was informed, [on reddit](https://www.reddit.com/r/rust/comments/6ynm53/a_simple_tip_to_improve_rust_program_speed/dmotelg/?context=3) that `rustc -C target-cpu=native` was a valid option, but I could not see it listed in the help menu.

## Meta

I see this issues on

```
rustc 1.22.0-nightly (088216fb9 2017-09-04) #current nightly
rustc 1.20.0 (f3d6973f4 2017-08-27) #stable
```

## How to reproduce

```
$ rustc -C target-cpu=help | grep native | wc -l
0
$ rustc -C target-cpu=native
error: no input filename given
```

### PR DESCRIPTION
**Pull request by mattico**
Fixes #44393

### CODE REVIEW

**Comment by mattico** [comment]
Bump

**Comment by arielb1** [comment]
Is there an easy way to add the *current* native CPU. i.e., to make the message look like:
```
"    native - Select the CPU of the current host (currently core2).
```

**Comment by arielb1** [comment]
I'm ok with r+-ing this PR as it is, but I think the change above could be an improvement.

**Comment by mattico** [comment]
Another idea: If we're cross compiling then we probably don't want to print this info, since it'll be wrong/misleading.

**Comment by mattico** [comment]
```
λ build\x86_64-pc-windows-msvc\stage1\bin\rustc -C target-cpu=help
Available CPUs for this target:
    native         - Select the CPU of the current host (currently haswell).
    amdfam10       - Select the amdfam10 processor.
    athlon         - Select the athlon processor.
<snip>
λ build\x86_64-pc-windows-msvc\stage1\bin\rustc -C target-cpu=help --target arm-unknown-linux-gnueabi
Available CPUs for this target:
    arm1020e      - Select the arm1020e processor.
    arm1020t      - Select the arm1020t processor.
<snip>
```

**Comment by kennytm** [comment]
Failed to build rustc_llvm on `dist-x86_64-linux alt`.

```
[00:07:21] cargo:warning=../rustllvm/PassWrapper.cpp: In function 'void LLVMRustPrintTargetCPUs(LLVMTargetMachineRef)':
[00:07:21] cargo:warning=../rustllvm/PassWrapper.cpp:310:55: error: cannot pass objects of non-trivially-copyable type 'std::string {aka class std::basic_string<char>}' through '...'
[00:07:21] cargo:warning=       MaxCPULen, "native", sys::getHostCPUName().str());
[00:07:21] cargo:warning=                                                       ^
[00:07:21] cargo:warning=../rustllvm/PassWrapper.cpp:310:55: warning: format '%s' expects argument of type 'char*', but argument 4 has type 'std::string {aka std::basic_string<char>}' [-Wformat=]
```

File: src/rustllvm/PassWrapper.cpp [hunk 1]
```diff
@@ -299,10 +299,15 @@ static size_t getLongestEntryLength(ArrayRef<SubtargetFeatureKV> Table) {
...
   const MCSubtargetInfo *MCInfo = Target->getMCSubtargetInfo();
+  const Triple::ArchType HostArch = Triple(sys::getProcessTriple()).getArch();
+  const Triple::ArchType TargetArch = Target->getTargetTriple().getArch();
   const ArrayRef<SubtargetFeatureKV> CPUTable = MCInfo->getCPUTable();
   unsigned MaxCPULen = getLongestEntryLength(CPUTable);
 
   printf("Available CPUs for this target:\n");
+  if (HostArch == TargetArch)
+    printf("    %-*s - Select the CPU of the current host (currently %s).\n",
+      MaxCPULen, "native", sys::getHostCPUName().data());
```
Comment (rkruppe) [inline src/rustllvm/PassWrapper.cpp]: StringRefs are not guaranteed to be null terminated. Use `%*` and pass the length as well, as in https://github.com/rust-lang/rust/blob/5dfc84cfa72b405c194228b53c4de3f6474204ec/src/rustllvm/PassWrapper.cpp#L651-L655

**Pull request merged**